src/cpu/flexcpu/flexcpu_thread.cc

```diff
@@ -345,6 +346,17 @@ FlexCPUThread::commitAllAvailableInstructions()
 
         inflightInsts.pop_front();
         buffer_shrunk = true;
+
+        if (!interrupts_checked
```

Comment:
Can you add a comment describing when interrupts are handled? Is this as soon as they are posted no matter what?